aten/src/ATen/native/Copy.cpp

```diff
@@ -207,7 +207,9 @@ static Tensor & copy_impl(Tensor & self, const Tensor & src, bool non_blocking)
   }
 
   if (src.is_meta()) {
-    TORCH_CHECK_NOT_IMPLEMENTED(false, "Cannot copy out of meta tensor; no data!")
+    TORCH_CHECK_NOT_IMPLEMENTED(
+        false,
+        "Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to().")
```

Comment:
I don't think it is ok to add this here. This error will happen for any copy of a Tensor into another.
It would be better for the Module code itself to catch this error, add the context and rethrow the error.
You can add it in https://github.com/pytorch/pytorch/blob/99ddfaf572d152e1fa741fe0f546f742c6452b2e/torch/nn/modules/module.py#L799 that is called by all the to(), cuda(), cpu(), etc methods. There you can make sure the error message is the one from this error, and if so, add the extra comment you have here.

Reply:
Thanks for the suggestion. Tried to add try,except,raise in `_apply` but there is no effect. The original error is throw 
https://github.com/pytorch/pytorch/blob/99ddfaf572d152e1fa741fe0f546f742c6452b2e/torch/nn/modules/module.py#L1150

Reply:
Tested locally
```
with torch.device('cuda'):
    m1 = torch.nn.Linear(10, 5)
m1.to("cpu")
```
working as is.

```
with torch.device('meta'):
    m2 = torch.nn.Linear(10, 5)
m2.to("cpu")
``` gives

```
Traceback (most recent call last):
  File ".\torch\nn\modules\module.py", line 1159, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
NotImplementedError: Cannot copy out of meta tensor; no data!

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 19, in <module>
    m2.to("cpu")
  File ".\torch\nn\modules\module.py", line 1166, in to
    return self._apply(convert)
           ^^^^^^^^^^^^^^^^^^^^
  File ".\torch\nn\modules\module.py", line 833, in _apply
    param_applied = fn(param)
                    ^^^^^^^^^
  File ".\torch\nn\modules\module.py", line 1162, in convert
    raise NotImplementedError(
NotImplementedError: Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to().
```

Reply:
```
with torch.device('meta'):
    m2 = torch.nn.Linear(10, 5)
m2.to("cpu")
``` 

gives

```
Traceback (most recent call last):
  File ".\torch\nn\modules\module.py", line 1159, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
NotImplementedError: Cannot copy out of meta tensor; no data!

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 19, in <module>
    m2.to("cpu")
  File ".\torch\nn\modules\module.py", line 1166, in to
    return self._apply(convert)
           ^^^^^^^^^^^^^^^^^^^^
  File ".\torch\nn\modules\module.py", line 833, in _apply
    param_applied = fn(param)
                    ^^^^^^^^^
  File ".\torch\nn\modules\module.py", line 1162, in convert
    raise NotImplementedError(
NotImplementedError: Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to().
```

The result is desired.

Reply:
The issue of error message happened twice is amended by 
```
                    raise NotImplementedError(
                        f"{e} Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to()."
                    ) from None
```